examples/deepcopy-gen/generators/deepcopy.go

```diff
@@ -745,6 +745,10 @@ func (g *genDeepCopy) doMap(t *types.Type, sw *generator.SnippetWriter) {
 	case uet.IsAssignable():
 		sw.Do("(*out)[key] = val\n", nil)
 	case uet.Kind == types.Interface:
+		// Note: do not generate code that won't compile as `DeepCopyinterface{}()` is not a valid function
+		if uet.Name.Name == "interface{}" {
+			klog.Fatalf("Hit an unsupported type %q", uet.Name.Name)
```

Comment:
I think I'd expect a runtime check as to whether it implements the deep copy interface rather than a hard fail?

@sttts

Reply:
@lavalamp Can you describe a bit more precisely what you have in mind please? I'm not sure I understand your comment.

Reply:
@lavalamp I think the change is fine. For every other, named interface you get proper output from the compiler which tells you that you have to implement the deep copy func. For `interface{}` though we get syntactical garbage. We could improve the former case too, but that's a different issue worth a follow-up (with limited value).

Reply:
SGTM